frontend/src/components/LoadData.tsx

```diff
@@ -426,7 +435,8 @@ export function LoadData(props: { children?: ReactNode }) {
                     }, [] as string[])
                     .join('/')
                   setter((map) => {
-                    const arr = map.get(key) || []
+                    const newMap = structuredClone(map)
```

Comment:
This is good because you are not supposed to mutate the atoms, but I think you can still reuse the contents, so a full clone is probably not necessary.
```suggestion
                    const newMap = {...map}
```

Reply:
ha--I tried {...map} first -- but it does a shallow clone--each map has an immutable array that also has to be cloned :(

Reply:
Ah, so recoil does lock everything inside the state as well. Very well.